### ISSUE TITLE
control installing of Rust packages

### ISSUE THREAD

**Issue by PetrKralCZ** [vscentrum/vsc-software-stack#83]
- For installing software written in Rust we are using the `cargo install` command. This command downloads all the latest dependencies.
- We need to have full control over the versions of dependencies being downloaded and ideally let EasyBuild download them itself and install them from local directories.
- The problem is, it seems the `cargo` command always expects there is a `Cargo.toml` file specifying these dependencies.
- The `cargo install <dependency_name>` only works for installing programs and not libraries.
```
error: there is nothing to install in `anyhow v1.0.68`, because it has no binaries
`cargo install` is only for installing programs, and can't be used with libraries.
To use a library crate, add it as a dependency in a Cargo project instead.
```
- The `cargo fetch command` also expects a `Cargo.toml` file
```
error: could not find `Cargo.toml` in `/home/user` or any parent directory
```
- The fact that there is a `--offline` option for the `cargo install` and the `cargo vendor` command suggests there should be a way to install dependencies from a local directory.
- Otherwise the `ext_list` of Rust-written dependencies would probably need to be used to generate a custom `Cargo.toml` file.

**Comment by boegel** [comment]
Hmm, maybe it would be better to have an issue for this in the EasyBuild GitHub repository, since this is not exactly specific to VSC...

Thoughts @smoors, @lexming?

**Comment by boegel** [comment]
https://github.com/easybuilders/easybuild-easyblocks/pull/2902 looks like an interesting start?

**Comment by boegel** [comment]
closing this since https://github.com/easybuilders/easybuild-easyblocks/pull/2902 was merged and included in EasyBuild v4.7.2
